server/middleware/userCheckMiddleware.test.ts

```diff
@@ -45,4 +46,97 @@ describe('createUserCheckMiddleware', () => {
     expect(handler).not.toHaveBeenCalled()
     expect(next).toHaveBeenCalledWith(new UnauthorizedError())
   })
+
+  describe('userIsAuthorisedForManage', () => {
+    it('returns middleware that calls next() with an unauthorized error when the path is not reports', async () => {
+      const handler = jest.fn()
+      const request = createMock<Request>({ path: paths.assessments.index({}) })
+      const response = createMock<Response>({
+        locals: {
+          user: userFactory.build({ roles: ['referrer'] }),
+        },
+      })
+      const next = jest.fn()
+
+      const middleware = createUserCheckMiddleware(userIsAuthorisedForManage)
+
+      middleware(handler)(request, response, next)
+
+      expect(handler).not.toHaveBeenCalled()
+      expect(next).toHaveBeenCalledWith(new UnauthorizedError())
+    })
+
+    describe('when the user has assessor role', () => {
+      it('returns middleware that calls the handler', async () => {
+        const handler = jest.fn()
+        const request = createMock<Request>({ path: paths.assessments.index({}) })
+        const response = createMock<Response>({
+          locals: {
+            user: userFactory.build({ roles: ['assessor'] }),
+          },
+        })
+        const next = jest.fn()
+
+        const middleware = createUserCheckMiddleware(userIsAuthorisedForManage)
+
+        middleware(handler)(request, response, next)
+
+        expect(handler).toHaveBeenCalledWith(request, response, next)
+      })
+    })
+
+    describe('when the user has reporter role', () => {
+      it('returns middleware that calls the handler when the the path is reports#new', async () => {
+        const handler = jest.fn()
+        const request = createMock<Request>({ path: paths.reports.new({}) })
+        const response = createMock<Response>({
+          locals: {
+            user: userFactory.build({ roles: ['reporter' as TemporaryAccommodationUserRole] }),
```

Comment:
Is this casting temporary whilst we await the type updates from the API repo?

Reply:
Exactly, I'll be creating a task in the backlog to remove once the types are updated.